calliope/core/model.py

```diff
@@ -92,47 +86,35 @@ def __init__(self, config, model_data=None, *args, **kwargs):
 
         self.plot = plotting.ModelPlotMethods(self)
 
-    def _init_from_model_run(self, model_run, debug_data):
+    def _init_from_model_run(self, model_run, debug_data, debug):
         self._model_run = model_run
-        self._debug_data = debug_data
         log_time(
             logger,
             self._timings,
             "model_run_creation",
             comment="Model: preprocessing stage 1 (model_run)",
         )
 
-        self._model_data_original = build_model_data(model_run)
+        model_data_factory = ModelDataFactory(model_run)
```

Comment:
Why do we call several methods on ModelDataFactory here? It seems like those should be called in ModelDataFactory.__init__ instead so that instantiating a model_data_factory gives you all the different data you need in one go, without the need to know about the internal methods of the class.

Reply:
It's split out for what I assume were the same reasons for splitting out the parts of the process previously: testing. If initialising the ModelDataFactory does *everything* in one go, then it's a pain to test some of the key intermediate steps in the process.

Reply:
So move what's in `__init__()` right now into a separate `_initial_init()` or something, then have `__init__()` call the four init functions in turn, but still test them separately?

Reply:
I agree with Stefan here. This approach seems error prone, because you can easily forget to call one of these methods.

If it's difficult to test that's usually a sign for a design flaw. As fixing the design may be too much here I'd go for a pragmatic, even if slightly ugly, way: call "private" methods in the test. In that case, you'd make all these methods private and call them from within a creation method (could be `build()`, or `create`, but I suggest `__call__`). Then you can test subsets of that creation method in your test by calling private methods instead of the creation method itself. 

The API then looks like:
```Python
model_data_factory = ModelDataFactory(model_run)
model_data = ModelDataFactory()
```

Reply:
I'm in favour of this change. However, calling ModelDataFactory would need to return four objects, not just `model_data`, which should be reserved as a pointer to the main xarray dataset. So it would be: 

```
model_data_factory = ModelDataFactory(model_run)
model_data_pre_clustering, model_data, data_pre_time, stripped_keys = ModelDataFactory()
```

Is that also OK?

Reply:
I think that's fine.

Reply:
OK, I've made that change, although I'm liking ModelDataFactory less and less at this point :P